src/main/java/com/st/utopia/agent/controller/SeatTickets.java

```diff
@@ -0,0 +1,189 @@
+	 */
+	private <T> ResponseEntity<T> methodCall(final String url, final HttpMethod method) {
+		return restTemplate.exchange(url, method, null,
+				new ParameterizedTypeReference<T>() {});
+	}
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for a method call.
+	 * @param url the URL to send the REST request to
+	 * @param type of method to send the REST request to
+	 * @param HttpEntity to send in the REST request
+	 * @param <T> the type we expect
+	 * @param <U> the type of the body sent to the server
+	 * @return the response the server sent
+	 */
+	private <T, U> ResponseEntity<T> methodCall(final String url, final HttpMethod method,
+			final HttpEntity<U> body) {
+		return restTemplate.exchange(url, method, body,
+				new ParameterizedTypeReference<T>() {});
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seats")
+	public ResponseEntity<SeatLocation> getAllSeatsWithPlan(
+			@PathVariable final String flightId) {
+		// FIXME: Search service doesn't yet provide this
+		String url = searchAPI + "/seats?flight=" + flightId;
+		return this.<SeatLocation>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seat/{row}/{seatId}")
+	public ResponseEntity<Ticket> getTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId) {
+		String url = bookingAPI + "/details/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+		return this.<Ticket>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Pay the price for ticket or extend the time
+	 * 
+	 * @return a ticket or an object
+	 */
+	@PutMapping("/flight/{flightId}/seat/{row}/{seatId}/ticket")
+	public ResponseEntity<Object> updateTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final PaymentAmount pay) {
+		// may need to just create a new object
+		if (pay.getPrice() == null) {
+			HttpEntity<PaymentAmount> body = new HttpEntity<>(pay);
+			String url = bookingAPI + "/pay/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+			return this.<Object, PaymentAmount>methodCall(url, HttpMethod.PUT, body);
+		} else {
+			String url = bookingAPI + "/extend/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+			return this.<Object>methodCall(url, HttpMethod.PUT);
+		}
+	}
+
+	/**
+	 * Book flight
+	 * 
+	 * @return a Ticket
+	 */
+	@PostMapping("/flight/{flightId}/seat/{row}/{seatId}/ticket")
+	public ResponseEntity<Ticket> postTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final User reserver) {
+		// not sure if the request body int worked
+		// it should through an error if they do not give a reserver (right?)
```

Comment:
You mean "throw" an error; "through" is not a verb. (But are these comments still applicable?)

Reply:
Yeah, I do not think they are applicable